components/file_attachment/index.ts

```diff
@@ -14,7 +15,8 @@ import FileAttachment from './file_attachment';
 
 function mapStateToProps(state: GlobalState) {
     const config = getConfig(state);
-    const pluginMenuItems = (state.plugins.components.FilesDropdown || []) as unknown as FileDropdownPluginComponent[];
+    const getPluginMenuItems = memoizeResult((state: GlobalState) => state.plugins.components.FilesDropdown || []);
```

Comment:
This actually won't work since it creates a new `getPluginMenuItems` every time this is called, and the new version doesn't remember the previous result. Since it's the same selector in both places, maybe it could go in `selectors/plugins.ts` and be shared between the two?

Reply:
Also, `memoizeResult` only returns the same result if the arguments are the same, so this'll still return a new array if `state` changes at all. Using `createSelector` would fix that since we could make it only recalculate if `state.plugins.components.FilesDropdown` changes like
```js
createSelector(
    (state) => state.plugins.components.FilesDropdown,
    (components) => {
        return components || [];
    },
);
```

Reply:
done @hmhealey